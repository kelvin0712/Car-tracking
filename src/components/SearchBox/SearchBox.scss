.SearchBox {
  display: flex;
  align-self: stretch;
  padding-left: 1rem;
  padding-top: 1rem;
  input[type=text] {
    padding: 8px;
    font-size: 13px;
    border: 1px solid grey;
    width: 50%;
    background: #f1f1f1;
    text-align: center;
  }

  button {
    padding: 8px;
    background: #2196F3;
    color: white;
    font-size: 13px;
    border: 1px solid grey;
    border-left: none;
    cursor: pointer;
  }
}