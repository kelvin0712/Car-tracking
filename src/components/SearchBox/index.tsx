import React from 'react';
import './SearchBox.scss';

export default class SearchBox extends React.Component<{
  onQuery(query: string): void
}> {
  inputRef = React.createRef<HTMLInputElement>()

  handleOnClick = () => {
    const { current } = this.inputRef
    if (current) this.props.onQuery(current.value)
  }

  render() {
    return (
    <div className='horizontal-layout'>
      <input type="text" ref={this.inputRef} />
      <button onClick={this.handleOnClick}>SEARCH</button>
    </div>
    )
  }
}
