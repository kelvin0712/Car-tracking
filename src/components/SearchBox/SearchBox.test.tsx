import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import SearchBox from '.';

it('Should match shapshot', () => {
  const { asFragment } = render(<SearchBox onQuery={() => {}} />)

  expect(asFragment()).toMatchSnapshot()
})


it('Should fire onQuery with the value of the input', () => {
  const onQuery = jest.fn() // spy

  const { container } = render(<SearchBox onQuery={onQuery} />)
  const input = container.querySelector('input')
  input.value = 'Test'

  fireEvent.change(input)
  const button = container.querySelector('button')
  fireEvent.click(button)

  expect(onQuery).toBeCalledWith('Test')
})
