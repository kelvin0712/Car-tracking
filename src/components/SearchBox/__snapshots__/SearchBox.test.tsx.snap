// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match shapshot 1`] = `
<DocumentFragment>
  <div
    class="SearchBox"
  >
    <input
      type="text"
    />
    <button>
      SEARCH
    </button>
  </div>
</DocumentFragment>
`;
